### PR TITLE
Multi table inheritance support

### PR DESCRIPTION
**Pull request by jacobtoppm**
Currently, non-Page models using multi table inheritance are not fully supported. When referencing the child model (eg via a foreign key) only the base model will be imported. This PR attempts to address this issue, treating it similarly to `Page` models by retrieving the specific instances when the serializer fetches them by id, using logic based on the approach used by Django-Model-Utils. This keeps the mappings and references using base models exclusively.

It also fixes an inconsistency in ChooserBlock, where child (not base) models were being added to mappings/references when a child model was referenced.

Thanks to @stevejalim for discovering these issues!

### CODE REVIEW

**Comment by stevejalim** [comment]
Not a code review, but I've tried out this branch with an MTI-based model (an MTI subclass of `wagtailmedia.Video`) and it does indeed now transfer the object, including field data that only exists within the subclass. 👍 👍 👍  Thanks for the rapid fix!

(Not sure if it's related, but I also noticed that tags on the video item did not transfer, but that could well just be a WIP configuration gap at my side)

File: wagtail_transfer/views.py [hunk 1]
```diff
@@ -126,8 +126,8 @@ def objects_for_export(request):
 
         for obj in serializer.get_objects_by_ids(ids):
             instance_serializer = get_model_serializer(type(obj))  # noqa
```
Comment (gasman) [inline wagtail_transfer/views.py:128]: I guess the `# noqa` can be removed now...

File: wagtail_transfer/serializers.py [hunk 2]
```diff
@@ -9,6 +11,65 @@
...
+        subclasses.append(rel.get_accessor_name())
+    return subclasses
+
+
+def _get_sub_obj_recurse(obj, s):
+    """
+    Given an object and its potential subclasses in lookup string form,
+    retrieve its most specific subclass recursively
+    Taken from: https://github.com/jazzband/django-model-utils/blob/master/model_utils/managers.py
+    """
```
Comment (gasman) [inline wagtail_transfer/serializers.py:45]: I'm sure the recursion here could be avoided just by splitting the string on LOOKUP_SEP and looping over the resulting list, but I guess it's not worth the extra overhead of writing new code :-)

**Review by gasman — APPROVED** [review]
Looks good to me!

**Pull request merged**